packages/devtools_app/lib/src/screens/memory/panes/diff/diff_pane.dart

```diff
@@ -15,9 +23,270 @@ class DiffPane extends StatefulWidget {
   State<DiffPane> createState() => _DiffPaneState();
 }
 
+/// Will notify if count of items or current index changed.
+class _DiffPaneController with ChangeNotifier {
+  _DiffPaneController() {
+    index.addListener(() => notifyListeners());
+  }
+
+  final scrollController = ScrollController();
+
+  /// The list contains one item that show information and all others
+  /// are snapshots.
+  final snapshots = <DiffListItem>[InformationListItem()];
+
+  final index = ValueNotifier<int>(0);
+
+  /// If true, some process is going on.
+  final isProcessing = ValueNotifier<bool>(false);
+
+  DiffListItem get selected => snapshots[index.value];
+
+  bool get hasSnapshots => snapshots.length > 1;
+
+  Future<void> takeSnapshot() async {
+    isProcessing.value = true;
+    final future = snapshotMemory();
+    snapshots.add(
+      SnapshotListItem(
+        future,
+        _nextNameNumber(),
+        currentIsolateName ?? '<isolate-not-detected>',
+      ),
+    );
+
+    notifyListeners();
+    await future;
+    final newElementIndex = snapshots.length - 1;
+    scrollController.autoScrollToBottom();
+    index.value = newElementIndex;
+    isProcessing.value = false;
+  }
+
+  Future<void> clearSnapshots() async {
+    snapshots.removeRange(1, snapshots.length);
+    index.value = 0;
+    notifyListeners();
+  }
+
+  int _nextNameNumber() {
```

Comment:
nit: replace all instances of 'nameNumber' with 'displayNumber' for more clarity